### ISSUE TITLE
[BUG] DeepDrills is scheduled one time extra on the same day

### ISSUE THREAD

**Issue by Samyak2** [chaos-genius/chaos_genius#830]
## Describe the bug

DeepDrills runs one extra time after setting up a KPI and configuring anomaly. This occurs approximately 2/3rd of the time.

## Explain the environment
- **Chaos Genius version**: v0.5.0
- **OS Version / Instance**: All
- **Deployment type**: All

## Current behavior

- Setup a few KPIs (at least 3 to reproduce) and configure anomaly for them
- After all the tasks have completed, go to the task monitor and search for "complete"
- Notice that DeepDrills has run 3 times
    - first run was during KPI setup - this was expected
    - one run was along with anomaly, right after it was configured - this was expected
    - the other run can occur sometime before or after anomaly run - not expected
- The scheduler logs show that the scheduler scheduled a DeepDrills task - it was not supposed to do so until the next day

## Expected behavior

DeepDrills should not have run three times.

## Screenshots

![image](https://user-images.githubusercontent.com/34161949/158561580-cacc2b19-b8a4-4653-abc0-324653a2ce0b.png)

## Additional context

On reproducing this with 10 KPIs, the following was observed:
- 3 KPIs ran correctly - DeepDrills ran only twice
- 7 KPIs showed the bug - DeepDrills ran thrice

On logging the inputs given to the scheduler it was found that the issue occurs in this scenario (notice the microsecond part of the timestamps):
- KPI was setup at 2022-03-16T06:55:09.**11111**
- DeepDrills was first scheduled at 2022-03-16T06:55:09.**22222**. This is the value of `last_scheduled_time_rca` in the next run
- Scheduler next runs at 2022-03-16T06:55:XX.**33333** (any time, but with a microsecond value greater than that of `last_scheduled_time_rca`)
    - `scheduled_time` will become 2022-03-16T06:55:09.**33333** which is greater than `last_scheduled_time_rca`. Since DeepDrills was not run on/after scheduled_time, it is scheduled again
    - This is because the microsecond part is not replaced along with the other time fields: https://github.com/chaos-genius/chaos_genius/blob/d64d0251388ec0d44b837c1e909643cad3a624a3/chaos_genius/jobs/analytics_scheduler.py#L49-L52

To summarize, the conditions under which this occurs is:
- DeepDrills first runs on the same second as when the KPI was created
- Scheduler runs at a time such that the microsecond component of the scheduler run time is greater than the value it was when DeepDrills last ran

## Solution/Fix

Add `microsecond=0` to all the `datetime.replace` function calls in scheduler. This will ensure that `scheduled_time` is always less than `last_scheduled_time_rca` if the date, hour, minute and second are the same.

## Logs

N/A

Thanks to @rjdp for helping debug, finding the root cause and testing the solution for this.
